experimental/native_file_system/virtual_root_provider_win.cc

```diff
@@ -1,12 +1,63 @@
-// Copyright (c) 2014 Intel Corporation. All rights reserved.
-// Use of this source code is governed by a BSD-style license that can be
-// found in the LICENSE file.
-
-#include "xwalk/experimental/native_file_system/virtual_root_provider.h"
-
-#include "base/logging.h"
-
-VirtualRootProvider::VirtualRootProvider() {
-  // TODO(darktears): Windows support to be added.
-  NOTIMPLEMENTED();
-}
+// Copyright (c) 2015 Intel Corporation. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "xwalk/experimental/native_file_system/virtual_root_provider.h"
+
+#include <shlobj.h>
+
+#include "base/files/file_path.h"
+#include "base/files/file_util.h"
+
+bool VirtualRootProvider::testing_enabled_ = false;
+
+VirtualRootProvider::VirtualRootProvider() {
+  if (!virtual_root_map_.empty())
+    return;
+
+  if (testing_enabled_)
```

Comment:
imho it would make more sense to make 'testing_enabled_' a constructor parameter, but I'm not insisting since it's apparently a part of the existing design/implementation..